serviceAccessConfig/lib/serviceAccessConfig/accessrulegenerator.py

```diff
@@ -0,0 +1,186 @@
+#  Copyright (C) 2016 SUSE LLC, Robert Schweikert <rjschwei@suse.com>
+#  All rights reserved.
+#
+#  This file is part of serviceAccessConfig
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Base class for all serviceAccessConfig plugins"""
+
+import ConfigParser
+import hashlib
+import logging
+import os
+import requests
+import sys
+import threading
+
+from generatorexceptions import *
```

Comment:
Don't let @schaefi see that; [he'll get upset.](https://github.com/SUSE/azurectl/pull/105/files/b2ade0de81ef386cc2e42dc67a842d345c059146#diff-b13dabd37635a09d6edef0e6c0ab4406R17) ;-)